File: app/src/main/java/org/metabrainz/mobile/data/sources/api/SearchService.java
package org.metabrainz.mobile.data.sources.api;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface SearchService {

    @GET("{entity}/")
    Call<ResponseBody> searchEntity(@Path("entity") String entity,
                                    @Query("query") String searchTerm,
                                    @Query("limit") String limit,
                                    @Query("offset") String offset);

}

File: app/src/main/java/org/metabrainz/mobile/data/sources/api/SearchService.kt
package org.metabrainz.mobile.data.sources.api

import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface SearchService {
    @GET("{entity}/")
    suspend fun searchEntity(@Path("entity") entity: String?,
                             @Query("query") searchTerm: String?,
                             @Query("limit") limit: Int,
                             @Query("offset") offset: Int): ResponseBody?
}